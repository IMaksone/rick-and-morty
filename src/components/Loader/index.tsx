import { ReactNode } from "react";

import classes from "./loader.module.css";

interface LoaderInterface {
  loaded: Boolean;
  children: ReactNode;
}

export default function Loader({ loaded, children }: LoaderInterface) {
  
  return<Spinner />
  // return loaded ? children : <Spinner />;
}

const Spinner = () => (
  <div className={classes.spiner_wr}>
    <div className={classes.spinner}></div>
  </div>
);
