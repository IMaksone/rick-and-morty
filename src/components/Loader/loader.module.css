.spiner_wr {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.spinner {
  display: block;
  position: relative;
  width: 56px;
  height: 56px;
  border-radius: 50%;
  border: 5px solid var(--light-color);
  clip: rect(0px, 150px, 150px, 75px);
  animation: rotate 1s linear infinite;
  -webkit-animation: rotate 1s linear infinite;
  -moz-animation: rotate 1s linear infinite;
}

.spinner::before {
  content: " ";
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  width: 66px;
  height: 0;
  background-color: var(--dark-color);
  animation: filling 2s linear infinite;
  -webkit-animation: filling 2s linear infinite;
  -moz-animation: filling 2s linear infinite;
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes filling {
  0% {
    height: 8px;
  }
  50% {
    height: 58px;
  }
  100% {
    height: 8px;
  }
}
