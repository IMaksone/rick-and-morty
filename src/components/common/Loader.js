import "./loader.css";

export function Loader() {
  return (
    <div className="loader-box">
      <div className="loader"></div>
    </div>
  );
}
