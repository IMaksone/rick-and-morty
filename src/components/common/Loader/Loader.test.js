import renderer from "react-test-renderer";
import { Loader } from "./Loader";

it("TEST Loader", () => {
  const component = renderer.create(<Loader />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
