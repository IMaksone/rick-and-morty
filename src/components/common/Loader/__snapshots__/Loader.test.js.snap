// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Loader 1`] = `
<div
  className="loader-box"
>
  <div
    className="loader"
  />
</div>
`;
